processing/src/main/java/org/apache/carbondata/processing/util/CarbonDataProcessorUtil.java

```diff
@@ -604,4 +606,33 @@ public static boolean isHeaderValid(String tableName, String header,
     }
     return dateformatsHashMap;
   }
+
```

Comment:
This reflection code is not required. Please check how load happens through dataframe in CarbonDataLoadRDD. I guess you can make use of same code.
And your iterator supposed to extend `InputIterator` of PR33.